client/landing/stepper/declarative-flow/internals/steps-repository/design-setup/hooks/use-recipe.ts

```diff
@@ -67,15 +69,22 @@ const useRecipe = (
 		[]
 	);
 
-	const preselectedDesign = useMemo(
-		() =>
-			allDesigns?.designs.find( ( design ) =>
-				design.is_virtual
-					? design.recipe?.slug === preselectedThemeSlug
-					: design.slug === preselectedThemeSlug
-			),
-		[ allDesigns ]
-	);
+	const { data: preselectedDesignDetails } = useStarterDesignBySlug( preselectedThemeSlug || '', {
+		enabled: !! preselectedThemeSlug,
+		select: ( design: Design ) => {
+			if ( disableCheckoutImmediately && design?.style_variations ) {
+				design.style_variations = [];
+			}
+
+			const descriptionArray = design.description?.split( '.' );
+			design.shortDescription =
+				descriptionArray && descriptionArray.length > 0 ? descriptionArray[ 0 ] + '.' : null;
```

Comment:
[Nitpick] This could be replaced with:

`      design.shortDescription = descriptionArray?.[0] ? descriptionArray[0] + '.' : null;
`

What happens if the description already ends with `.`?